base/printf.jl

```diff
@@ -16,14 +16,15 @@ function gen(s::AbstractString)
             push!(blk.args, :(write(out, $(length(x)==1 ? x[1] : x))))
         else
             c = lowercase(x[end])
-            f = c=='f' ? gen_f :
-                c=='e' ? gen_e :
-                c=='a' ? gen_a :
-                c=='g' ? gen_g :
-                c=='c' ? gen_c :
-                c=='s' ? gen_s :
-                c=='p' ? gen_p :
-                         gen_d
+            if c=='f'; f = gen_f
+            elseif c=='e'; f = gen_e
```

Comment:
This is probably even more readable as:

```
f = if (c == 'f') gen_f
    elseif (c == 'e') gen_e
    ...
    else gen_d
end
```

Note: the paren are superfluous (just like the `;`).